src/inject_key_bin.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+
+#
+# Use this script to inject your own private key and authentication counter
+# into U2F binary. Might be useful if you want keys to survive firmware updates.
+#
+# Example:
+#
+# Generate EC private key with openssl:
+# > openssl ecparam -name prime256v1 -genkey -noout -outform der > key.der
+#
+# Inject generated key into u2f.bin and set auth counter to 100:
+# > python3 inject_key_bin.py --key key.der --ctr 100 --bin build/u2f.bin
+#
+# key will not be modified if --key parameter is not present
+# counter will not be modified if --ctr parameter is not present
+
+from __future__ import print_function
+from asn1crypto.keys import ECPrivateKey
+import hashlib
+import argparse
+import sys
+import struct
+import os
+
+# TODO: detect correct offset from .bin file.
+stm32f103_offset = 0xB800  # See stm32f103.ld
+#efm32hg_offset = ????
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--bin", default="build/u2f.bin",
+                    help='.bin file to inject keys into. Or "stdin"')
+parser.add_argument("--key", help="EC private key in DER format")
+parser.add_argument("--ctr", default=0, type=lambda x: int(x,0), help="Value of auth counter")
+parser.add_argument("--offset", default=stm32f103_offset, type=lambda x: int(x,0), help="Offset within file to patch")
```

Comment:
So, for stm32f103 offset can be 0xf400 or 0xf800 depending on build flags (CUSTOM_ATTESTATION_CERT), for efm32hg it can be 0xb400 or 0xb800. But it is more reliable to count backwards. It should always be 2048 bytes from the end of .bin file.

Reply:
Pushed new code that uses this calculation by default